tests/lit-tests/3229.ispc

```diff
@@ -0,0 +1,23 @@
+// Check argument type deduction for template functions with short vector types
```

Comment:
I also suggest to check short vectors of enums and bools and deduction in nested templates.

Reply:
I have added tests for bool and nested templates, but it looks like short vectors with enums as the type or size are not supported.

Reply:
Ah, ok. It should be easy to add it but it is out of scope of this PR.